modules/grpc/common/credentials/grpc-credentials-builder.cpp

```diff
@@ -271,7 +277,14 @@ ClientCredentialsBuilder::build() const
     case GCAM_ALTS:
       return ::grpc::experimental::AltsCredentials(alts_credentials_options);
     case GCAM_ADC:
+    {
+      if (!application_credentials.empty())
+        {
+          if (setenv("GOOGLE_APPLICATION_CREDENTIALS", application_credentials.c_str(), 1) != 0)
+            msg_info("gRPC: setenv failed for GOOGLE_APPLICATION_CREDENTIALS");
+        }
       return ::grpc::GoogleDefaultCredentials();
+    }
```

Comment:
We should store the old value and restore it after the `GoogleDefaultCredentials` call.